src/helpers/testContextBuilder.js

```diff
@@ -87,14 +87,15 @@ function buildTestContextFromJump(testContext, testMap, jump) {
  * @returns {Object} the new test context
  * @private
  */
-function getTestContext(testContext, testMap, item, section, part, position) {
+function getTestContext(testContext, testMap, item, section, part, position, attempt) {
     var isLeavingSection = section.id !== testContext.sectionId,
         isLeavingPart = part.id !== testContext.testPartId,
         newTestContext = _.defaults(
             {
                 itemIdentifier: item.id,
                 itemPosition: position,
                 remainingAttempts: Math.max(-1, item.remainingAttempts - 1),
+                attempt: attempt ? attempt + 1 : 1,
```

Comment:
Maybe should you only set the attempt if provided. Otherwise, it is safer to not add it and keep the value from the test context

Reply:
fixed